Mathlib/MeasureTheory/Function/LpSeminorm.lean

```diff
@@ -1341,48 +1360,53 @@ theorem snorm_le_snorm_top_mul_snorm (p : ℝ≥0∞) (f : α → E) {g : α →
     exact ha
   by_cases hp_zero : p = 0
   · simp only [hp_zero, snorm_exponent_zero, MulZeroClass.mul_zero, le_zero_iff]
-  simp_rw [snorm_eq_lintegral_rpow_nnnorm hp_zero hp_top, snorm_exponent_top, snorm_ess_sup]
+  simp_rw [snorm_eq_lintegral_rpow_nnnorm hp_zero hp_top, snorm_exponent_top, snormEssSup]
   calc
-    (∫⁻ x, ↑‖b (f x) (g x)‖₊ ^ p.to_real ∂μ) ^ (1 / p.to_real) ≤
-        (∫⁻ x, ↑‖f x‖₊ ^ p.to_real * ↑‖g x‖₊ ^ p.to_real ∂μ) ^ (1 / p.to_real) := by
+    (∫⁻ x, (‖b (f x) (g x)‖₊ : ℝ≥0∞) ^ p.toReal ∂μ) ^ (1 / p.toReal) ≤
+        (∫⁻ x, (‖f x‖₊ : ℝ≥0∞) ^ p.toReal * (‖g x‖₊ : ℝ≥0∞) ^ p.toReal ∂μ) ^ (1 / p.toReal) := by
```

Comment:
Are all these ` : ℝ≥0∞)`s necessary? I thought the point in the elaboration trick above at the top of the file to make them not so!

Reply:
We need these type ascriptions in either case.
The elaboration trick is due to `MeasureTheory.Functions.LpSpace`, actually, not required in this file.
This was misleading so I've removed.

Reply:
@kmill, do you think writing an elaborator to make these go away is sensible/worth the effort?

Certainly there's no need to do it before this PR is merged.